telemetry/interface.go

```diff
@@ -0,0 +1,54 @@
+// Package telemetry provides instrumentation tools for other modules and packages.
+package telemetry
+
+import (
+	"time"
+
+	gometrics "github.com/armon/go-metrics"
+)
+
+// Metrics defines a wrapper around application telemetry functionality. It allows
+// metrics to be gathered at any point in time. When creating a Metrics object,
+// internally, a global metrics is registered with a set of sinks as configured
+// by the operator. In addition to the sinks, when a process gets a SIGUSR1, a
+// dump of formatted recent metrics will be sent to STDERR.
+type Metrics interface {
```

Comment:
nit: It's a bit weird to define a "generic" metrics interface where some of the methods have return values that use a specific metric's package types, i.e. `github.com/armon/go-metrics`. What do you think about this? Can we convert these return types to interfaces? i.e. `gometrics.Label`

Reply:
yes, we can, 
actually, I'd like to extend the data types to provide better instrumentation, but that will be a huge PR, so I started with a small one.

Reply:
Let's create an issue and add a TODO reference in the godocs prior to merging this 🙏